frontend/src/components/TaskCreator/DatePicker.tsx

```diff
@@ -65,142 +100,130 @@ export default function DatePicker(props: Props): ReactElement {
     );
   };
 
-  /**
-   * Whether or not this task is a repeating task.
-   */
-  const [isRepeat, setIsRepeat] = React.useState<boolean>(!(date instanceof Date));
-
   /**
    * Event handler for when the user starts the repeat box
    */
   const changeRepeat = (e: ChangeEvent): void => {
-    setIsRepeat((e.target as HTMLSelectElement).value === 'true');
+    setInternalDate({
+      ...internalDate,
+      type:
+       (e.target as HTMLSelectElement).value === 'true'
```

Comment:
There are definitely better way than this cast.
Usually I use `e.currentTarget.value` for this. `ChangeEvent` type can be rewritten as `ChangeEvent< HTMLSelectElement>`.